src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -118,15 +162,120 @@ else if (this.superType != null)
      * @param name  Path of the registry's {@link ResourceLocation}
      * @param sup   Supplier of the RegistryBuilder that is called to fill {@link #type} during the NewRegistry event
      * @return      A supplier of the {@link IForgeRegistry} created by the builder.
+     *
+     * @deprecated Use {@link #create(ResourceLocation, String)} and {@link #makeRegistry(Class, Supplier)} instead
      */
+    @Deprecated(forRemoval = true, since = "1.18.2")
     public Supplier<IForgeRegistry<T>> makeRegistry(final String name, final Supplier<RegistryBuilder<T>> sup) {
-        if (this.superType == null)
-            throw new IllegalStateException("Cannot create a registry without specifying a base type");
-        if (this.type != null || this.registryFactory != null)
-            throw new IllegalStateException("Cannot create a registry for a type that already exists");
+        return makeRegistry(new ResourceLocation(modid, name), this.superType, sup);
+    }
+
+    /**
+     * Only used for custom registries to fill the forge registry held in this DeferredRegister.
+     *
+     * Calls {@link RegistryBuilder#setName} and {@link RegistryBuilder#setType} automatically.
+     *
+     * @param base The base type to use in the created {@link IForgeRegistry}
+     * @param sup Supplier of a RegistryBuilder that initializes a {@link IForgeRegistry} during the {@link NewRegistryEvent} event
+     * @return A supplier of the {@link IForgeRegistry} created by the builder.
+     * Will always return null until after the {@link NewRegistryEvent} event fires.
+     */
+    public Supplier<IForgeRegistry<T>> makeRegistry(final Class<T> base, final Supplier<RegistryBuilder<T>> sup) {
```

Comment:
Formatting: braces on newlines.
```suggestion
    public Supplier<IForgeRegistry<T>> makeRegistry(final Class<T> base, final Supplier<RegistryBuilder<T>> sup)
    {
```